### PR TITLE
[PATCH] Limit the backoff time on retries

### PR DESCRIPTION
**Pull request by eb-enzo**
Limit the backoff time on retries so it doesn't exceeds the timeout
setting for the request.

The default of 10 retries can block the sender for up to around 128
seconds. As the default timeout is 5 seconds, it the request should
timeout quite before that.

### CODE REVIEW

File: pysoa/common/transport/redis_gateway/core.py [hunk 1]
```diff
@@ -347,7 +347,22 @@ def send_message(
             # Try at least once, up to queue_full_retries times, then error
             for i in range(-1, self.queue_full_retries):
                 if i >= 0:
-                    time.sleep((2 ** i + random.random()) / self.EXPONENTIAL_BACK_OFF_FACTOR)
+                    backoff_time_in_seconds = (2 ** i + random.random()) / self.EXPONENTIAL_BACK_OFF_FACTOR
```
Comment (brutledge-eb) [inline pysoa/common/transport/redis_gateway/core.py:350]: wonder what that random is for?

**Review by brutledge-eb — APPROVED** [review]
LGTM.

File: pysoa/common/transport/redis_gateway/core.py (hunk 1 above)
Comment (eb-enzo) [reply pysoa/common/transport/redis_gateway/core.py:350]: You don't want all the clients retrying at the same time I guess. So the random part will help will that.

**Comment by eb-enzo** [comment]
As the CI is broken, I will wait until it's fixed before retrying this fix.

**Pull request closed without merging**